dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdMappingFileUploadTask.kt

```diff
@@ -179,5 +193,61 @@ open class DdMappingFileUploadTask
         return true
     }
 
+    private fun shrinkMappingFile(
+        mappingFile: File
+    ): File {
+        if (!mappingFileTrimIndents && mappingFilePackagesAliases.isEmpty()) return mappingFile
+
+        val shrinkedFile = File(mappingFile.parent, "mapping-shrinked.txt")
+        // sort is needed to have predictable replacement in the following case:
+        // imagine there are 2 keys - "androidx.work" and "androidx.work.Job", and the latter
+        // occurs much often than the rest under "androidx.work.*". So for the more efficient
+        // compression we need first to process the replacement of "androidx.work.Job" and only
+        // after that any possible prefix (which has a smaller length).
+        val replacements = mappingFilePackagesAliases.entries
+            .sortedByDescending { it.key.length }
+            .map {
+                Regex("${it.key}(?=\\W)") to it.value
+            }
+
+        mappingFile.readLines()
+            .map {
+                if (mappingFileTrimIndents) it.trimStart() else it
+            }
+            .map {
+                applyShortAliases(it, replacements)
+            }
+            .forEach {
+                shrinkedFile.appendText(it + "\n")
+            }
+
+        return shrinkedFile
+    }
+
+    private fun applyShortAliases(
+        line: String,
+        replacements: List<Pair<Regex, String>>
+    ): String {
+        val isLineWithRename = !line.startsWith(MAPPING_FILE_COMMENT_CHAR) &&
+            line.contains(MAPPING_FILE_CHANGE_DELIMITER)
+
+        return if (isLineWithRename) {
+            val (source, obfuscated) = line.split(MAPPING_FILE_CHANGE_DELIMITER)
+            val aliasedSource =
+                replacements.fold(source) { state, entry ->
+                    val (from, to) = entry
+                    state.replace(from, to)
```

Comment:
do we have benefits/use case allowing more than 1 replacement per line ?

Reply:
We need to do that because it can be several types as function arguments mentioned on the single line, so several replacements can apply, ex.:

> 39:39:void ModalBottomSheetLayout-BzaUkTc(kotlin.jvm.functions.Function3,androidx.compose.ui.Modifier,androidx.compose.material.ModalBottomSheetState,androidx.compose.ui.graphics.Shape,float,long,long,long,kotlin.jvm.functions.Function2,androidx.compose.runtime.Composer,int,int):392:392 -> a